flake.nix

```diff
@@ -1,51 +1,41 @@
 {
   description = "cpiotools";
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.1";
 
-  outputs =
-    { self
-    , nixpkgs
-    , ...
-    } @ inputs:
+  outputs = inputs:
     let
-      nameValuePair = name: value: { inherit name value; };
-      genAttrs = names: f: builtins.listToAttrs (map (n: nameValuePair n (f n)) names);
-      allSystems = [ "x86_64-linux" "aarch64-linux" "i686-linux" "x86_64-darwin" "aarch64-darwin" ];
+      lastModifiedDate = inputs.self.lastModifiedDate or inputs.self.lastModified or "19700101";
+      version = "${builtins.substring 0 8 lastModifiedDate}-${inputs.self.shortRev or "dirty"}";
 
-      forAllSystems = f: genAttrs allSystems (system: f {
+      allSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+
+      forAllSystems = f: inputs.nixpkgs.lib.genAttrs allSystems (system: f {
         inherit system;
-        pkgs = import nixpkgs { inherit system; };
+        pkgs = import inputs.nixpkgs { inherit system; };
       });
     in
     {
-      devShell = forAllSystems ({ system, pkgs, ... }: self.packages.${system}.package.overrideAttrs ({ nativeBuildInputs ? [ ], ... }: {
+      devShell = forAllSystems ({ system, pkgs, ... }: inputs.self.packages.${system}.default.overrideAttrs ({ nativeBuildInputs ? [ ], ... }: {
```

Comment:
Should we keep using this, or consider switching to `devShells.${system}.default` while we're here?

Reply:
I've opted to just "modernize" the flake instead and bring it more in line with our others